### PR TITLE
Fix sign-up email template for emails with no breaches

### PR DESCRIPTION
**Pull request by flozia**
# References: 
Jira: [MNTOR-1258](https://mozilla-hub.atlassian.net/browse/MNTOR-1258)

# Description

Fix the generation of the sign-up email template for users that sign up with an email that is not involved in a relevant breach.

# Screenshot

mntor-1258-screenshot

# How to test

1. Sign up with an email address that has no relevant associated breaches.
2. Check out the email that we send upon sign-up.

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [ ] I've added a unit test to test for potential regressions of this bug.
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/views/emails/email-signup-report.js [hunk 1]
```diff
@@ -51,40 +51,50 @@ const signupReportEmailPartial = data => {
...
+              : getMessage('fxm-warns-you-no-breaches')
           }
         </p>
-        <table style='${breachSummaryTableStyle}'>
-          <tr>
-            <td>
-              ${emailBreachStats?.map(breachStat => `
-                <table style='${breachSummaryCardStyle}'>
+        ${
+          emailBreachStats?.length > 0
```
Comment (Vinnl) [inline src/views/emails/email-signup-report.js:69]: And likewise.

**Review by Vinnl — COMMENTED** [review]
The contents of the email indeed now indicate that there was no breach, however, the subject is still "Firefox Monitor found your email in a new data breach."

File: src/controllers/auth.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ async function confirmed (req, res, next, client = FxAOAuthClient) {
 
     // Send report email
     const utmCampaignId = 'report'
-    const heading = unsafeBreachesForEmail?.length
+    const heading = unsafeBreachesForEmail?.length > 0
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/controllers/auth.js]: Ups — right. Fixed by https://github.com/mozilla/blurts-server/commit/d6021f34e946a088d1dbdf07005be083ecea545d.

File: src/controllers/auth.js [hunk 3]
```diff
@@ -110,22 +110,21 @@ async function confirmed (req, res, next, client = FxAOAuthClient) {
 
     // Send report email
     const utmCampaignId = 'report'
-    const heading = unsafeBreachesForEmail?.length
+    const subject = unsafeBreachesForEmail?.length
```
Comment (flozia) [inline src/controllers/auth.js:113]: > The contents of the email indeed now indicate that there was no breach, however, the subject is still "Firefox Monitor found your email in a new data breach."

@Vinnl The dynamic subject was mistakenly switched with the heading.

**Review by Vinnl — APPROVED** [review]
I wasn't able to test this with an already-breached primary account because, as mentioned on Slack, I get an error about `breachData` being undefined at this line after signing up and being redirected back to Monitor: https://github.com/mozilla/blurts-server/blob/2e38da45327898f22ce49fe7022bf916138bbd6c/src/views/emails/email-breach-card.js#L55

But for a non-breached account, I now get the correct email subject.

**Comment by flozia** [comment]
> I wasn't able to test this with an already-breached primary account because, as mentioned on Slack, I get an error about `breachData` being undefined …

Seems like I’ve only tested the sign-up with a non-breached account. Should be fixed by [d450050](https://github.com/mozilla/blurts-server/pull/2860/commits/d450050f7418436fb533c3cf74ecb5b467f8be78).

**Comment by mansaj** [comment]
> I wasn't able to test this with an already-breached primary account because, as mentioned on Slack, I get an error about `breachData` being undefined at this line after signing up and being redirected back to Monitor:
> 
> https://github.com/mozilla/blurts-server/blob/2e38da45327898f22ce49fe7022bf916138bbd6c/src/views/emails/email-breach-card.js#L55
> 
> But for a non-breached account, I now get the correct email subject.

I was able to test signup with an email that already has breaches. I got the correct email with the following subject line:
`Firefox Monitor found your info in these breaches`

**Review by mansaj — APPROVED** [review]

**Pull request merged**